src/routes/acquirerRoutes.ts

```diff
@@ -123,6 +145,16 @@ router.post("/:resourceID/:step", async (req: Request, res: Response) => {
     return res.status(400).send("Step data not found");
   }
 
+  const { day, month, year } = req.body;
+
+    if(!isValidDate(day, month, year)) {
+        res.render(`../views/acquirer/${formStep}.njk`, {
+            errorMessage: "Please enter a valid date."
+            // could add specific input field validation messages e.g: "Please enter valid Year"
+        });
+      return;
+    }
+
```

Comment:
I think it might be better to split the validation out into a separate function, so when we need to do validation for other steps we don't end up clogging up the `router.post` function.

We could have a separate validation function, something like:
```
const validateRequestBody = (step: string, body: RequestBody) => {
  let errorMessage = "";
  
  switch (step) {
    case 'date':
      const { day, month, year } = body;
      if (!isValidDate(day, month, year)) {
        errorMessage = "Please enter a valid date."
      }
      break;
  }

  return errorMessage
}
```
which we can add more cases to for other steps.

Then in the `router.post` we can validate the body, keep track of any errors in the session and redirect back to the original page to display the errors on the form, e.g:
```
  const errorMessage = validateRequestBody(formStep, req.body)
  stepData.errorMessage = errorMessage;
  if (errorMessage) {
    return res.redirect(`/acquirer/${resourceID}/${formStep}`)
  }
```